lib/scsi.h

```diff
@@ -416,67 +414,44 @@ static_assert(sizeof(Read10Command) == 9);
 // SCSI Reference Manual Table 99
 // https://www.seagate.com/files/staticfiles/support/docs/manual/Interface%20manuals/100293068j.pdf
 struct Read12Command {
+  uint8_t obsolete : 2;
+  bool rarc : 1;  // Rebuild Assist Recovery bit
+  bool fua : 1;   // Forced Unit access bit
+  bool dpo : 1;   // disable page output bit
   uint8_t rd_protect : 3;
-  bool dpo : 1;     // disable page output bit
-  bool fua : 1;     // Forced Unit access bit
-  bool rarc : 1;    // Rebuild Assist Recovery bit
-  bool fua_nv : 1;  // Forced Access Unit Non-Volatile
-  bool obsolete : 1;
   uint32_t logical_block_address : 32;
   uint32_t transfer_length : 32;
-  bool restricted_mmc_6 : 1;
-  uint8_t reserved : 2;
   uint8_t group_number : 5;
+  uint8_t reserved : 2;
+  bool restricted_mmc_6 : 1;
   ControlByte control_byte;
 } ABSL_ATTRIBUTE_PACKED;
 static_assert(sizeof(Read12Command) == 11);
 
 // SCSI Reference Manual Table 100
 // https://www.seagate.com/files/staticfiles/support/docs/manual/Interface%20manuals/100293068j.pdf
 struct Read16Command {
+  bool dld_2 : 1;
+  bool obsolete : 1;
+  bool rarc : 1;  // Rebuild Assist Recovery bit
+  bool fua : 1;   // Forced Unit access bit
+  bool dpo : 1;   // disable page output bit
   uint8_t rd_protect : 3;
-  bool dpo : 1;     // disable page output bit
-  bool fua : 1;     // Forced Unit access bit
-  bool rarc : 1;    // Rebuild Assist Recovery bit
-  bool fua_nv : 1;  // Forced Access Unit Non-Volatile
-  bool reserved1 : 1;
   uint64_t logical_block_address : 64;
   uint32_t transfer_length : 32;
-  bool restricted_mmc_6 : 1;
-  uint8_t reserved2 : 2;
-  uint8_t group_number : 5;
+  uint8_t group_number : 6;
+  bool dld_0 : 1;
+  bool dld_1 : 1;
```

Comment:
This section has a bigger than normal change; group_number changed size, and the mmc field was removed entirely.  This is in addition to the reserved field being replaced with dld.

Reply:
Yeah, while I was making the changes just for the bit field ordering, I realized that some of the structs followed the SPC-3 specs instead of SPC-4 so I made them all consistent with SPC-4

Reply:
Sounds good, thanks for clarifying.